src/main/java/Duke.java

```diff
@@ -51,14 +66,31 @@ public static void main(String[] args) {
                         index = -1;
                     }
                     todoList.markItem(index, false);
+                }else if(func.equals("todo")){
+                    Todo todo = Todo.toTodo(instruction);
+                    if(todo == null){
+                        printError("Wrong todo format");
+                        continue;
+                    }
+                    todoList.addItem(todo);
+                }else if(func.equals("deadline")){
+                    Deadline deadline = Deadline.toDeadline(instruction);
+                    if(deadline == null){
+                        printError("Wrong deadline format");
+                        continue;
+                    }
+                    todoList.addItem(deadline);
+                }else if(func.equals("event")){
+                    Event event = Event.toEvent(instruction);
+                    if(event == null){
+                        printError("Wrong event format");
+                        continue;
+                    }
+                    todoList.addItem(Event.toEvent(instruction));
```

Comment:
There is deep nesting in **if-else** statement while every **if** statements inside are conducting similar calculation. There would be other way to better construct this code like declaring special function for those calculation.